File viewer content : version 2

Natika31 opened the pull request:
- Séparation du fichier et de son contenu dans deux classes distinctes. Tests mis à jours et passés avec succès.
- Questions :  
- >  respect de l'architecture en couches => est-il obligatoire de créer une classe service réservée aux op CRUD pour le contenu du fichier ou bien peut-on directement réaliser ces opérations depuis la classe service du fichier? 
- > bonnes pratiques > setters => je ne sais pas trop où réaliser les manips qui permettent de lire et récupérer le contenu du fichier en hexadécimal (utilisation des classes File, FileInputStream...). Si je le mets dans la méthode "set_binary_content" de la classe FileContent, c'est ok? Le constructeur et la méthode  "set_binary_content" prendraient alors en paramètre le chemin d'origine du fichier à uploader, ce qui me parait plus simple que de mettre le contenu hexa en paramètre. 
- > bonnes pratiques > nommage =>j'ai nommé l'attribut qui contient le chemin d'origine du fichier à uploader  (chemin système du fichier original) "file_origin_location" mais ça me parait pas terrible. Comment peut-on nommer simplement cet attribut sans  créer d’ambiguïté? Une suggestion?

Natika31 committed:
replace field name "fileRepository" with "fileDao"

Natika31 committed:
création de la couche de services

Natika31 committed:
refactor file preloading

Natika31 committed:
save a non-empty directory

Natika31 committed:
rename fileController, fileService and fileServiceImpl

Natika31 committed:
enregistement du contenu du fichier dans une table différente de la
table du fichier

KevinDelcourt commented:
Je répond mais je t'avoue que j'ai vraiment pas envie d'argumenter.

1. La 1ère option c'est mieux

2. Vaut mieux pas toucher au setter je crois que ça peut perturber Spring. Tu peux peut-être faire ça dans la couche service c'est mieux.

3. file_origin_path ? Par contre si ça représente le chemin dans le disque dure de l'utilisateur qui upload le fichier dans ton appli c'est bizarre de le sauvegarder

Sinon le code ne m'a pas choqué mais je suis passé dessus très très vite

Natika31 committed:
refactoring couche service

- Separation fileContentService de fileService
- Opération de conversion du contenu binaire du fichier en entree  dans
couche service
- Renommage "file_origin location" en "file_origin_path"
